pyicloud/services/drive.py

```diff
@@ -260,6 +262,11 @@ def date_last_open(self):
 
     def open(self, **kwargs):
         """Gets the node file."""
+        # iCloud returns 400 Bad Request for 0-byte files
+        if self.data["size"] == 0:
+            response = Response()
+            response.raw = io.BytesIO()
+            return response
```

Comment:
In which case 0 bit file exists ?

Why should we respond with an empty response or not raise an exception ?

Reply:
In my case it was some autogenerated log file that was yet to be filled. A zero-byte file is a valid file (eg. `touch file` on a system) so it shouldn't raise an exception (as it currently does). There was also a PR (#294) to allow uploading zero-byte files - you'd need this PR to actually download them.